futures-util/src/stream/stream/split.rs

```diff
@@ -81,20 +98,14 @@ impl<S: Sink<Item>, Item> Sink<Item> for SplitSink<S, Item> {
     fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), S::Error>> {
         let this = &mut *self;
         let mut inner = ready!(this.lock.poll_lock(cx));
-        if this.slot.is_some() {
-            ready!(inner.as_pin_mut().poll_ready(cx))?;
-            inner.as_pin_mut().start_send(this.slot.take().unwrap())?;
-        }
+        ready!(Self::poll_flush_slot(inner.as_pin_mut(), &mut this.slot, cx))?;
```

Comment:
Move the `this.lock.poll_lock(cx)` out of `poll_flush_slot ` to avoid the need to `poll_lock` again to get the `inner`.